Client/Frontend/Accessors/HomePageAccessors.swift

```diff
@@ -26,8 +26,14 @@ class HomePageAccessors {
 
     static func isButtonEnabled(_ state: AppState) -> Bool {
         switch state.ui {
-        case .tab:
-            return true
+        case .tab(let tabState):
+            if hasHomePage(state) {
+                return true
+            } else if let url = tabState.url {
+                return url.isValid
+            } else {
+                return false
```

Comment:
You can remove the `else` conditions here and shorten this to `return tabState.url?.isValid ?? false`.